src/main/java/reposense/authorship/FileInfoExtractor.java

```diff
@@ -109,20 +110,36 @@
                 continue;
             }
 
+            if (!isValidAndNonBinaryFile(filePath, nonBinaryFilesList)) {
+                continue;
+            }
+
             if (Format.isInsideWhiteList(filePath, config.getFormats())) {
-                try {
-                    FileInfo currentFileInfo = generateFileInfo(config.getRepoRoot(), filePath);
-                    setLinesToTrack(currentFileInfo, fileDiffResult);
-                    fileInfos.add(currentFileInfo);
-                } catch (InvalidPathException ipe) {
-                    logger.warning(String.format(INVALID_FILE_PATH_MESSAGE_FORMAT, filePath));
-                }
+                FileInfo currentFileInfo = generateFileInfo(config.getRepoRoot(), filePath);
+                setLinesToTrack(currentFileInfo, fileDiffResult);
+                fileInfos.add(currentFileInfo);
             }
         }
 
         return fileInfos;
     }
 
+    /**
+     * Returns a {@code Set} of non-binary files for the repo {@code repoConfig}.
+     */
+    public static Set<Path> getListOfNonBinaryFiles(RepoConfiguration repoConfig) {
+        String receivedMsg = GitDiff.gitGetModifiedFiles(Paths.get(repoConfig.getRepoRoot()));
+        String[] listOfFiles = receivedMsg.split("\n");
```

Comment:
Same as the comment on `message`, for `receivedMsg`, use a more appropriate name.